src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -4144,8 +4199,7 @@ create_py_vnodelist(pbs_list_head *vnlist)
 				goto create_py_vnodelist_exit;
 			}
 
-			rc = pbs_python_populate_python_class_from_svrattrl(\
-						       py_vn, &rqs.rq_attr);
+			rc = pbs_python_populate_python_class_from_svrattrl(py_vn, &rqs.rq_attr, perf_label, perf_action);
```

Comment:
Wouldn't this also call hook_perf_stat_start and hook_perf_stat_stop with the same perf_label and perf_action as you created in 4101?
`create_py_vnodelist`:4101 - `hook_perf_stat_start` - creates a new p_stat and sets initial values
`pbs_python_populate_python_class_from_svr_attrl`:2279 - `hook_perf_stat_start` - finds the existing p_stat and overwrites values
`pbs_python_populate_python_class_from_svr_attrl`:2336 - `hook_perf_stat_stop` - finds the p_stat, stops it, prints a summary, frees p_stat
`create_py_vnodelist`:4275 - `hook_perf_stat_stop` - can't find the p_stat because it's been freed, returns without printing.

Reply:
Good catch, Vincent! I was supposed to NULL the values of  'perf_label' and 'perf_action' parameters to the inner call of pbs_pbs_python_populate_python_class_from_svrattrl(), so perf stat collecting task does not happen. That function is being called for each object created by create_vnodelist(), create_joblist() and create_resvlist(), and that could be a lot (e.g. ex 10,000 vnodes!) which would fill the daemon logs too much. So I opted to just do 1 perf stat perf_hook_start at the beginning, and 1 perf_hook_stop at end of create_vnodelist(), create_joblist(), and create_resvlist(). So a call to create_vnodelist() would give a single elapsed time for how long it took say 10,000 vnodes to get created, rather than 10,000 separate daemon log entries. I just missed NULL-ing the perf_* parameters to *populate_class_from_svrattrl()...